gnomad/utils/vcf.py

```diff
@@ -92,3 +345,536 @@ def get_pipe_expr(array_expr: hl.expr.ArrayExpression) -> hl.expr.StringExpressi
+    :param bool verbose: If True, show top values of annotations being checked, including checks that pass; if False,
+        show only top values of annotations that fail checks.
+    :param str subpop: Subpop abbreviation, supplied only if subpopulations are included in the annotation groups being checked.
+    :param sort_order: List containing order to sort label group combinations. Default is SORT_ORDER.
+    :rtype: None
+    """
+    combo_AC = [ht.info[f"{prefix}AC_{x}"] for x in make_label_combos(label_groups)]
+    combo_AN = [ht.info[f"{prefix}AN_{x}"] for x in make_label_combos(label_groups)]
+    combo_nhomalt = [
+        ht.info[f"{prefix}nhomalt_{x}"] for x in make_label_combos(label_groups)
+    ]
+
+    group = label_groups.pop("group")[0]
+    alt_groups = "_".join(
+        sorted(label_groups.keys(), key=lambda x: sort_order.index(x))
+    )
+
+    annot_dict = {
+        f"sum_AC_{group}_{alt_groups}": hl.sum(combo_AC),
+        f"sum_AN_{group}_{alt_groups}": hl.sum(combo_AN),
+        f"sum_nhomalt_{group}_{alt_groups}": hl.sum(combo_nhomalt),
+    }
+
+    ht = ht.annotate(**annot_dict)
+
+    for subfield in ["AC", "AN", "nhomalt"]:
+        if not subpop:
+            generic_field_check(
+                ht,
+                (
+                    ht.info[f"{prefix}{subfield}_{group}"]
+                    != ht[f"sum_{subfield}_{group}_{alt_groups}"]
+                ),
+                f"{prefix}{subfield}_{group} = sum({subfield}_{group}_{alt_groups})",
+                [
+                    f"info.{prefix}{subfield}_{group}",
+                    f"sum_{subfield}_{group}_{alt_groups}",
+                ],
+                verbose,
+            )
+        else:
+            generic_field_check(
+                ht,
+                (
+                    ht.info[f"{prefix}{subfield}_{group}_{subpop}"]
+                    != ht[f"sum_{subfield}_{group}_{alt_groups}"]
+                ),
+                f"{prefix}{subfield}_{group}_{subpop} = sum({subfield}_{group}_{alt_groups})",
+                [
+                    f"info.{prefix}{subfield}_{group}_{subpop}",
+                    f"sum_{subfield}_{group}_{alt_groups}",
+                ],
+                verbose,
+            )
+
+
+def set_female_y_metrics_to_na(
+    t: Union[hl.Table, hl.MatrixTable], faf: bool = True,
+) -> Dict[str, hl.expr.Int32Expression]:
+    """
+    Set AC, AN, and nhomalt chrY variant annotations for females to NA (instead of 0).
+
+    :param t: Table/MatrixTable containing female variant annotations.
+    :return: Dictionary with reset annotations
+    """
+    metrics = list(t.row.info)
+    female_metrics = [x for x in metrics if "_female" in x]
+    female_metrics = [
+        x for x in female_metrics if ("nhomalt" in x) or ("AC" in x) or ("AN" in x)
+    ]
+
+    if faf:
+        female_metrics.extend([x for x in female_metrics if "faf" in x])
```

Comment:
I'm confused by this line: it looks like you're extending the existing list `female_metrics` by faf elements that are already in the list

Reply:
faf elements aren't in the list -- `female_metrics` first pulls all annotations that contain `_female` and then corrects that list to keep only elements with both `_female` and `AC`, `AN`, or `nhomalt`. the faf annotations (`faf95_female`) don't contain AC, AN, or nhomalt, so they aren't kept in that step.

that said, why do we set AC, AN, and nhomalt to 0 but leave AF? I'm going to change this to just be `female_metrics = [x for x in metrics if "_female" in x]` if that works with you and @jkgoodrich

Reply:
hmm, interesting that AF is not set to 0. I will defer to @gtiao for this one, there may have been motivation for this that I am not aware of.